code/modules/mob/living/silicon/robot/sprites/combat.dm

```diff
@@ -6,10 +6,16 @@
 
 	var/has_speed_sprite = FALSE
 	var/has_shield_sprite = FALSE
+	var/has_shield_and_speed = FALSE //RS Edit
```

Comment:
I don't think we should add even more booleans.... rather convert those has_xxx _sprite into a flag